### PR TITLE
Cu 1wg23uy allow for multiline input for postings brian gao

### PR DESCRIPTION
**Pull request by BENJIGAO**
### [CLICKUP TICKET LINK](https://app.clickup.com/t/r17re8)
- Created multiline functionality for input boxes for postings. For example, when specifying requirements of a postings, before you had to manually input them one-by-one through a dialog box. Now, you can use bullet points to input them all at once.
- This was done because the business team would like the functionality of copy and pasting a list (e.g., from Google docs) into the CMS for ease of use, which wasn't possible before.
- Utilized draft.js component for the rich-text/multiline functionality
- Created rich-text utility functions to convert arrays to `EditorState` (the object required by draft.js components) and vice versa. This was necessary because the backend stores arrays, whereas the frontend stores an `EditorState` object
- Refactored the backend db functions to fit the new functionality. Before, we only had an insert and delete function that could only handle one entity/object at a time. Now, it can replace many entities.
- Deleted now unused code, such as dialog boxes and old db functions
- Used Prettier to automatically reformat some files

Note: Go through the PR commit-by-commit. Although there are a lot of changed lines, this is mainly due to Prettier.

## Type of change
- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)

# How Has This Been Tested?
- [x] Checked "/example" endpoint on postman
- [x] Ran frontend-backend test to ensure that saving changes updates the backend accordingly

To test the functionality, enter the `Current Openings` section of the CMS. Then edit a posting by pressing `Edit`:

image

Change the `Requirements`, `Additional Info`, `Tasks`, `Recommended Skills`, and `Skills to be Learned` to whatever you want. However, it's important to specify them as a bulleted list, where each bullet represent one entity that will be stored in the database. For example, if you want `React`, and `Redux` as requirements, then you would specify them in the picture shown below:

image

Once you are done, press `Save` at the bottom, which saves your changes. To see your changes, you can either enter the same posting in the frontend of you can check the backend through Postman by querying the posting with its ID:

image

# Checklist:

- [x] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation/READ-ME/UMLS (if applicable)
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Comment by jennchenn** [comment]
Task linked: [CU-1wg23uy Allow for multiline input for postings](https://app.clickup.com/t/1wg23uy)

File: src/frontend/pages/postings/forms/EditPostingsForm.js [hunk 1]
```diff
@@ -55,7 +55,6 @@ const EditPostingsForm = () => {
     title,
     teamId,
     deadline,
-    // location,
```
Comment (SteveShengStar) [inline src/frontend/pages/postings/forms/EditPostingsForm.js:58]: You can delete this line.

File: src/frontend/pages/postings/forms/EditPostingsForm.js [hunk 2]
```diff
@@ -317,7 +306,6 @@ const EditPostingsForm = () => {
       {reqNotFilled && (
         <RequiredText>Please fill out all required fields.</RequiredText>
       )}
-      {renderAddNewDialog()}
```
Comment (SteveShengStar) [inline src/frontend/pages/postings/forms/EditPostingsForm.js:320]: Are you sure we don't need this anymore ? What was the purpose of this dialog ?

**Comment by BENJIGAO** [comment]
Can someone test this on Safari? I tested it on Microsoft Edge and Firefox and it works fine.

File: src/frontend/pages/postings/forms/EditPostingsForm.js (hunk 1 above)
Comment (BENJIGAO) [reply src/frontend/pages/postings/forms/EditPostingsForm.js:58]: I believe it was removed originally 😅

File: src/frontend/pages/postings/forms/EditPostingsForm.js (hunk 2 above)
Comment (BENJIGAO) [reply src/frontend/pages/postings/forms/EditPostingsForm.js:325]: I believe we don't need this anymore. This function was used to render a dialog, so for example, when you wanted to add a new requirement, you would press the "Add New+" button to generate the following dialog:
image
However, now that we using multiline inputs for each of these fields, there's no more use for a dialog and thus this function as well. I also deleted all the other dialog functions

**Review by jmawloo — APPROVED** [review]

**Pull request merged**